tools/clang/lib/SPIRV/SpirvBuilder.cpp

```diff
@@ -369,19 +369,25 @@ SpirvUnaryOp *SpirvBuilder::createUnaryOp(spv::Op op, QualType resultType,
                                           SpirvInstruction *operand,
                                           SourceLocation loc,
                                           SourceRange range) {
+  if (!operand)
+    return nullptr;
```

Comment:
How does this happen? This is a change in behaviour, so I want to understand why.

Reply:
So when we have an error, we call emittError and then return null.
Example: when handling an non-implemented intrinsic function, if the function is not implemented, we emit, and return null.

But the expression is still getting parsed. The code usually is fine with that, the null moves around, and in the end, we return nullptr and show the error.

But this means functions like that can get called with nullptr.
The exact chain for this bug is:
 - castToInt(loadIfGL(subExpr)).

The loadIfGL is done on a subExpr using a non-implemented intrinsic (dot4), and returns nullptr. castToInt calls doUnaryOp, because it's a cast, but with a null operand.

Initially, I thought I should fix this call-site, but the pattern is a bit everywhere.

What I think should be good is:
 - in the SpirvBuilder, assert for null parameters.
 - in the emitter, allow null params, but check for validity before doing ptr->toRValue() but also when calling spirvBuilder functions like this one.

Reply:
The ideal solution is to emit and error before we start translating the AST into spir-v, but making the code more robust is good too.